Build warning; object size

I see a singular build warning while building with GCC 12.3 (on NixOS, if that matters):

```
In file included from /nix/store/n8dryz4xf7ln028j37zapgzg4j47p743-gcc-12.3.0/include/c++/12.3.0/bits/stl_uninitialized.h:63,
                 from /nix/store/n8dryz4xf7ln028j37zapgzg4j47p743-gcc-12.3.0/include/c++/12.3.0/memory:65,
                 from include/boost/asio/detail/memory.hpp:21,
                 from include/boost/asio/execution/detail/as_invocable.hpp:20,
                 from include/boost/asio/execution/execute.hpp:23,
                 from include/boost/asio/execution/executor.hpp:25,
                 from include/boost/asio/execution/allocator.hpp:20,
                 from include/boost/asio/execution.hpp:18,
                 from include/boost/asio/any_completion_executor.hpp:22,
                 from include/boost/asio.hpp:20,
                 from /build/source/daemon/interface.cpp:21:
In static member function 'static _Tp* std::__copy_move<_IsMove, true, std::random_access_iterator_tag>::__copy_m(const _Tp*, const _Tp*, _Tp*) [with _Tp = unsigned char; bool _IsMove = false]',
    inlined from '_OI std::__copy_move_a2(_II, _II, _OI) [with bool _IsMove = false; _II = unsigned char*; _OI = unsigned char*]' at /nix/store/n8dryz4xf7ln028j37zapgzg4j47p743-gcc-12.3.0/include/c++/12.3.0/bits/stl_algobase.h:495:30,
    inlined from '_OI std::__copy_move_a1(_II, _II, _OI) [with bool _IsMove = false; _II = unsigned char*; _OI = unsigned char*]' at /nix/store/n8dryz4xf7ln028j37zapgzg4j47p743-gcc-12.3.0/include/c++/12.3.0/bits/stl_algobase.h:522:42,
    inlined from '_OI std::__copy_move_a(_II, _II, _OI) [with bool _IsMove = false; _II = unsigned char*; _OI = unsigned char*]' at /nix/store/n8dryz4xf7ln028j37zapgzg4j47p743-gcc-12.3.0/include/c++/12.3.0/bits/stl_algobase.h:529:31,
    inlined from '_OI std::copy(_II, _II, _OI) [with _II = unsigned char*; _OI = unsigned char*]' at /nix/store/n8dryz4xf7ln028j37zapgzg4j47p743-gcc-12.3.0/include/c++/12.3.0/bits/stl_algobase.h:620:7,
    inlined from 'std::pair<std::array<unsigned char, 6>, std::__cxx11::basic_string<char> > get_interface_mac(const std::string&)' at /build/source/daemon/interface.cpp:84:12:
/nix/store/n8dryz4xf7ln028j37zapgzg4j47p743-gcc-12.3.0/include/c++/12.3.0/bits/stl_algobase.h:431:30: warning: 'void* __builtin_memcpy(void*, const void*, long unsigned int)' writing 8 bytes into a region of size 6 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-overflow=-Wstringop-overflow=8;;]
  431 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/daemon/interface.cpp: In function 'std::pair<std::array<unsigned char, 6>, std::__cxx11::basic_string<char> > get_interface_mac(const std::string&)':
/build/source/daemon/interface.cpp:64:26: note: destination object 'mac' of size 6
   64 |   std::array<uint8_t, 6> mac{0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
      |                          ^~~
```

I'm not immediately sure how to quiet it, but since the project has otherwise perfect warning hygiene, it seems a shame to leave this one!

Unfortunately, same warning with the new form (it's line 65 because the original line is commented, not removed):

```
...
/nix/store/n8dryz4xf7ln028j37zapgzg4j47p743-gcc-12.3.0/include/c++/12.3.0/bits/stl_algobase.h:431:30: warning: 'void* __builtin_memcpy(void*, const void*, long unsigned int)' writing 8 bytes into a region of size 6 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-overflow=-Wstringop-overflow=8;;]
  431 |             __builtin_memmove(__result, __first, sizeof(_Tp) * _Num);
      |             ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/daemon/interface.cpp: In function 'std::pair<std::array<unsigned char, 6>, std::__cxx11::basic_string<char> > get_interface_mac(const std::string&)':
/build/source/daemon/interface.cpp:65:26: note: destination object 'mac' of size 6
   65 |   std::array<uint8_t, 6> mac{{0x00, 0x00, 0x00, 0x00, 0x00, 0x00}};
      |                          ^~~
```

I also tried with just:

```c++
std::array<uint8_t, 6> mac{};
```

and same deal :(

--- a/daemon/interface.cpp
+++ b/daemon/interface.cpp
@@ -81,7 +81,7 @@ std::pair<std::array<uint8_t, 6>, std::string> get_interface_mac(
   close(fd);
 
   uint8_t* sa = reinterpret_cast<uint8_t*>(ifr.ifr_hwaddr.sa_data);
-  std::copy(sa, sa + 8, std::begin(mac));
+  std::copy(sa, sa + 6, std::begin(mac));
 
   char str_mac[18];
   sprintf(str_mac, "%.2x:%.2x:%.2x:%.2x:%.2x:%.2x", mac[0], mac[1], mac[2],

